src/webviews/extension-side/dataviewer/dataVIewerDependencyService.ts

```diff
@@ -0,0 +1,162 @@
+        // From https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-pip (%conda is here as well).
+        return isConda ? '%conda' : '%pip';
+    }
+
+    public async checkAndInstallMissingDependencies(options: IDataViewerDependencyServiceOptions): Promise<void> {
+        const kernel = options.kernel;
+        const interpreter = options.interpreter;
+
+        // Providing feedback as soon as possible.
+        if (!kernel && interpreter) {
+            sendTelemetryEvent(Telemetry.InterpreterNotSupportedOnTheWeb);
+            throw new Error(DataScience.interpreterNotSupportedOnTheWeb());
+        }
+        if (!kernel) {
+            sendTelemetryEvent(Telemetry.InsufficientParameters);
+            throw new Error(DataScience.insufficientParameters());
+        }
+        if (kernel && !kernel.session) {
+            sendTelemetryEvent(Telemetry.NoActiveKernelSession);
+            throw new Error(DataScience.noActiveKernelSession());
+        }
+
+        const pandasVersion = await this.getVersionOfPandas(options);
+
+        if (pandasVersion) {
+            if (isVersionOfPandaSupported(pandasVersion)) {
+                return;
+            }
+            sendTelemetryEvent(Telemetry.PandasTooOld);
+            // Warn user that we cannot start because pandas is too old.
+            const versionStr = `${pandasVersion.major}.${pandasVersion.minor}.${pandasVersion.build}`;
+            throw new Error(DataScience.pandasTooOldForViewingFormat().format(versionStr));
+        }
+
+        sendTelemetryEvent(Telemetry.PandasNotInstalled);
+        await this.installMissingDependencies(options);
+    }
+
+    private async getVersionOfPandas(options: IDataViewerDependencyServiceOptions): Promise<SemVer | undefined> {
+        const kernel = options.kernel;
+
+        if (!kernel) {
+            return undefined;
+        }
+        const command = kernelGetPandasVersion;
+
+        try {
+            const outputs = await this.executeSilently(command, options);
+            return outputs.map((text) => (text ? parseSemVer(text.toString()) : undefined)).find((item) => item);
+        } catch (e) {
+            traceWarning(DataScience.failedToGetVersionOfPandas(), e.message);
+            return;
+        }
+    }
+
+    private async installMissingDependencies(options: IDataViewerDependencyServiceOptions): Promise<void> {
+        sendTelemetryEvent(Telemetry.PythonModuleInstall, undefined, {
+            action: 'displayed',
+            moduleName: ProductNames.get(Product.pandas)!
+        });
+
+        const kernel = options.kernel;
+
+        let selection: string | undefined;
+        if (kernel) {
+            selection = this.isCodeSpace
+                ? Common.install()
+                : await this.applicationShell.showErrorMessage(
+                      DataScience.pandasRequiredForViewing(),
+                      { modal: true },
+                      Common.install()
```

Comment:
If I remove the check for `this.isCodeSpace` and run `showErrorMessage`, it does work in the web, but only in this scenario, after installing pandas, I need to restart the kernel to make it work. If I keep the `this.isCodeSpace` check, I do not need to restart the kernel after `pandas` gets installed. Does this make sense? It is weird to me. I wonder if I'm doing something wrong.